ublox/src/ubx_packets/cfg_val.rs

```diff
@@ -236,15 +295,19 @@ macro_rules! cfg_val {
       }
 
       #[track_caller]
-      pub fn parse(buf: &[u8]) -> Self {
+      pub fn parse(buf: &[u8]) -> Option<Self> {
         let key_id = u32::from_le_bytes([buf[0], buf[1], buf[2], buf[3]]);
         match key_id {
           $(
             $cfg_key_id => {
-              Self::$cfg_item(from_cfg_v_bytes!(&buf[4..], $cfg_value_type))
+              Some(Self::$cfg_item(from_cfg_v_bytes!(&buf[4..], $cfg_value_type)))
             },
           )*
-          _ => unimplemented!("unknown key ID: 0x{:8X}", key_id),
+          _ => {
+            // TODO: add a mechanism to log such messages that supports also no_std
+            // eprintln!("unknown key ID: 0x{:8X}", key_id);
+            None
+           },
```

Comment:
This suggest that the function should actually return `Result` instead of `Option`. If you have an unknown key ID isn't that just an error?

Reply:
I think you are right. The `parse` function seems to be called only in the `CfgValIter` `next` method so at that level, Result can be handled the same as the Option.

Reply:
good point but a little confusing...